sdk/core/Azure.Core/src/Shared/ClientDiagnostics.cs

```diff
@@ -22,15 +22,16 @@ internal class ClientDiagnostics : DiagnosticScopeFactory
 
         private readonly HttpMessageSanitizer _sanitizer;
 
-        public ClientDiagnostics(ClientOptions options)
+        public ClientDiagnostics(ClientOptions options, bool? suppressNestedClientActivities = null)
                     : this(options.GetType().Namespace!,
                     GetResourceProviderNamespace(options.GetType().Assembly),
-                    options.Diagnostics)
+                    options.Diagnostics,
+                    suppressNestedClientActivities)
         {
         }
 
-        public ClientDiagnostics(string optionsNamespace, string? providerNamespace, DiagnosticsOptions diagnosticsOptions)
-            : base(optionsNamespace, providerNamespace, diagnosticsOptions.IsDistributedTracingEnabled, false)
+        public ClientDiagnostics(string optionsNamespace, string? providerNamespace, DiagnosticsOptions diagnosticsOptions, bool? suppressNestedClientActivities = null)
+            : base(optionsNamespace, providerNamespace, diagnosticsOptions.IsDistributedTracingEnabled, suppressNestedClientActivities.GetValueOrDefault(false))
         {
             // TODO: use DiagnosticsOptions.SuppressNestedClientSpans  here after new Azure.Core ships:
```

Comment:
nit: remove

Reply:
To make sure I understand: this will now be enabled once DPG passes `suppressNestedClientActivities = true` in ClientDiagnostics constructor? (per https://github.com/ShivangiReja/autorest.csharp/pull/2/files#diff-812ba0c844167e48b314a283eceeae710d5d35de9d3f758f61d3e6a6e2574f2f).  We no longer have a dependency on Core shipping, correct?

Reply:
Correct!